src/ConcordiumNetSdk/Transactions/AccountTransactionSignature.cs

```diff
@@ -0,0 +1,79 @@
+using System.Collections.Immutable;
+
+namespace ConcordiumNetSdk.Transactions;
+
+/// <summary>
+/// Signature of an account transaction.
+/// An account transaction signature is a map from indices of credentials to maps from key indices to signatures.
+/// A credential index is relative to the account address and a key index is relative to the credential.
+/// The number of credentials and keys per credential should be at least 1 and at most 255.
+/// </summary>
+public class AccountTransactionSignature
+{
+    /// <summary>
+    /// Internal representation of the signature. This is a map from account credential indices to account signature maps.
+    /// </summary>
+    public readonly ImmutableDictionary<byte, AccountSignatureMap> signature;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="AccountTransactionSignature"/> class.
+    /// </summary>
+    /// <param name="signatureMaps"><see cref="Dictionary"> corresponding to the map.</param>
+    private AccountTransactionSignature(Dictionary<byte, Dictionary<byte, byte[]>> signatureMaps)
+    {
+        var accountTransactionSignature = new Dictionary<byte, AccountSignatureMap>();
+        foreach (var m in signatureMaps)
+        {
+            var accountSignatureMap = new Dictionary<byte, byte[]>();
+            foreach (var k in m.Value)
+            {
+                accountSignatureMap.Add(k.Key, k.Value);
+            }
+            accountTransactionSignature.Add(m.Key, new AccountSignatureMap(accountSignatureMap));
+        }
+        this.signature = accountTransactionSignature.ToImmutableDictionary();
+    }
+
+    /// <summary>
+    /// Creates an account transaction signature by signing the provided data with the specified transaction signer.
+    /// </summary>
```

Comment:
It would be good to document preconditions/what exceptions are possible.

Reply:
This is implemented with a `Sign` method in `ITransactionSigner`/`TransactionSigner` rather than `Create`. Should anything be stated there?